core/templates/dev/head/pages/topic_editor/questions/QuestionsTabDirective.js

```diff
@@ -29,13 +29,18 @@ oppia.directive('questionsTab', [
         'EditableQuestionBackendApiService', 'EditableSkillBackendApiService',
         'MisconceptionObjectFactory', 'QuestionObjectFactory',
         'QuestionSuggestionObjectFactory', 'SuggestionThreadObjectFactory',
-        'EVENT_QUESTION_SUMMARIES_INITIALIZED', 'StateEditorService', function(
+        'EVENT_QUESTION_SUMMARIES_INITIALIZED',
+        'QuestionUndoRedoService', 'UrlService',
+        'UndoRedoService', function(
             $scope, $http, $q, $uibModal, $window, AlertsService,
             TopicEditorStateService, QuestionCreationService,
             EditableQuestionBackendApiService, EditableSkillBackendApiService,
             MisconceptionObjectFactory, QuestionObjectFactory,
             QuestionSuggestionObjectFactory, SuggestionThreadObjectFactory,
-            EVENT_QUESTION_SUMMARIES_INITIALIZED, StateEditorService) {
+            EVENT_QUESTION_SUMMARIES_INITIALIZED,
+            QuestionUndoRedoService, UrlService,
+            UndoRedoService) {
+          $scope.QuestionUndoRedoService = QuestionUndoRedoService;
```

Comment:
Is this used in the HTML file? If not, no need to have it in $scope.

Reply:
It is used in the html file now

Reply:
Ok, but even then it only uses that single function. So, just surface that function to scope instead of the whole service.

Reply:
done